docker-compose.yml

```diff
@@ -0,0 +1,211 @@
+version: "3.9"
+
+services:
+  # contracts
+
+  poa-ganache:
+    image: trufflesuite/ganache:v7.1.0
+    command: --wallet.deterministic --wallet.totalAccounts 50
+    ports:
+      - "8546:8545"
+    healthcheck:
+      test: ["CMD", "node", "-e", "require('http').request('http://localhost:8545').end()"]
+      interval: 5s
+      timeout: 5s
+    deploy:
+      mode: global
+
+  poa-deploy-contracts:
+    build: contracts
+    command: npx truffle migrate --to 4 --network audius_dev
+    depends_on:
+      poa-ganache:
+        condition: service_healthy
+
+  # eth-contracts
+
+  eth-ganache:
+    image: trufflesuite/ganache:v7.1.0
+    command: --wallet.deterministic --wallet.totalAccounts 50 --miner.blockTime 1
+    ports:
+      - "8545:8545"
+    healthcheck:
+      test: ["CMD", "node", "-e", "require('http').request('http://localhost:8545').end()"]
+      interval: 5s
+      timeout: 5s
+    deploy:
+      mode: global
+
+  eth-blockscout-db:
+    image: postgres:13.6
+    environment:
+      POSTGRES_PASSWORD: 'postgres'
+      POSTGRES_USER: 'postgres'
+    healthcheck:
+      test: ["CMD", "pg_isready"]
+      interval: 10s
+      timeout: 5s
+    deploy:
+      mode: global
+
+  eth-blockscout:
+    image: blockscout/blockscout:latest
+    command: 'mix do ecto.create, ecto.migrate, phx.server'
+    environment:
+      ETHEREUM_JSONRPC_VARIANT: 'ganache'
+      ETHEREUM_JSONRPC_HTTP_URL: 'http://eth-ganache:8545'
+      ETHEREUM_JSONRPC_WS_URL: 'ws://eth-ganache:8545'
+      INDEXER_DISABLE_PENDING_TRANSACTIONS_FETCHER: 'true'
+      DATABASE_URL: 'postgresql://postgres:postgres@eth-blockscout-db:5432/postgres?ssl=false'
+      ECTO_USE_SSL: false
+    ports:
+      - "9545:4000"
+    depends_on:
+      eth-blockscout-db:
+        condition: service_healthy
+      eth-ganache:
+        condition: service_healthy
+
+  eth-deploy-contracts:
+    build: eth-contracts
+    command: npx truffle migrate --to 12 --network audius_dev
+    depends_on:
+      eth-ganache:
+        condition: service_healthy
+
+  # solana-programs
+
+  solana-test-validator:
+    image: solanalabs/solana:v1.10.11
+    entrypoint: solana-test-validator
+    command: --quiet --gossip-host solana-test-validator
+    ports:
+      - "8899:8899"
```

Comment:
can we add a comment here about why this needs 4 pairs of ports mapped?

Reply:
This was introduced in https://github.com/AudiusProject/audius-protocol/pull/1786/files#diff-a88334f7ae0bc40f3ab3982d5f623c1d3bac1e9e1422e4ff2a23d5f69384b8afL51
cc: @hareeshnagaraj

Reply:
Only one port is mapped now 🎉